pkg/distributor/query.go

```diff
@@ -19,15 +20,15 @@ import (
 )
 
 // Query multiple ingesters and returns a Matrix of samples.
-func (d *Distributor) Query(ctx context.Context, from, to model.Time, matchers ...*labels.Matcher) (model.Matrix, error) {
+func (d *Distributor) Query(ctx context.Context, userID string, from, to model.Time, matchers ...*labels.Matcher) (model.Matrix, error) {
```

Comment:
Shouldn't we enforce the limits here as well?

Reply:
Yes; my focus was on things that blow up, which a range query is much more likely to.

Reply:
The `Query()` is used for range queries too, no? It's used when the "gRPC streaming" is disabled, while "QueryStream()" is called when it's enabled.

Reply:
Oh, sorry, I got confused.

For non-streaming queries and chunks, ingester enforces the limits already.

Reply:
> For non-streaming queries and chunks, ingester enforces the limits already.

Where is it done? I can't find it.